stdlib/tests/collections/smt.rs

```diff
@@ -1,664 +1,194 @@
-
-    // insert a value under key_c which has the same 32-bit prefix as A and B; this inserts
-    // one entry into the advice map
-    let raw_c = 0b00000000_00000000_11111111_11111111_00111111_01111111_11111111_11111111_u64;
-    let key_c = RpoDigest::from([ONE, ONE, ONE, Felt::new(raw_c)]);
-    let val_c = [ONE, ONE, ONE, ZERO];
-
-    let init_smt = smt.clone();
-    smt.insert(key_c.into(), val_c);
-    let new_map_entries = [build_node_entry(key_c, val_c, 48)];
-    assert_set(&init_smt, key_c, EMPTY_VALUE, val_c, smt.root().into(), &new_map_entries);
-
-    // at this point the tree has 3 nodes A, B, C, all 3 are a depth 48 and share the same 32-bit
-    // prefix; also B and C share the same 34-bit prefix.
-
-    // remove node A from the tree; since B and C share the same 34-bit prefix, they remain at
-    // depth 48
-    let init_smt = smt.clone();
-    smt.insert(key_a.into(), EMPTY_VALUE);
-    assert_set(&init_smt, key_a, val_a2, EMPTY_VALUE, smt.root().into(), &[]);
-
-    // remove node B from the tree; this will move node C to depth 16, and thus inserts a new
-    // entry into the advice map for node C at depth 16
-    let init_smt = smt.clone();
-    smt.insert(key_b.into(), EMPTY_VALUE);
-    let new_map_entries = [build_node_entry(key_c, val_c, 16)];
-    assert_set(&init_smt, key_b, val_b, EMPTY_VALUE, smt.root().into(), &new_map_entries);
+    let key = LEAVES[0].0;
+    let value = [42323_u64.into(); 4];
+    let (init_stack, final_stack, store, advice_map) = prepare_insert_or_set(key, value, &mut smt);
+    build_test!(source, &init_stack, &[], store, advice_map).expect_stack(&final_stack);
 }
 
+/// Tests inserting an empty value to an empty tree
 #[test]
-fn tsmt_set_48_from_32() {
-    let mut smt = TieredSmt::default();
-
-    // insert a value under key_a into an empty tree
-    let raw_a = 0b00000000_00000000_11111111_11111111_11111111_11111111_11111111_11111111_u64;
-    let key_a = RpoDigest::from([ONE, ONE, ONE, Felt::new(raw_a)]);
-    let val_a = [ONE, ZERO, ZERO, ZERO];
-    smt.insert(key_a.into(), val_a);
+fn test_smt_set_empty_value_to_empty_tree() {
+    let mut smt = Smt::new();
+    let empty_tree_root = smt.root();
 
-    // insert a value under key_b which has the same 16-bit prefix as A
-    let raw_b = 0b00000000_00000000_01111111_11111111_01111111_11111111_11111111_11111111_u64;
-    let key_b = RpoDigest::from([ONE, ONE, ONE, Felt::new(raw_b)]);
-    let val_b = [ONE, ONE, ZERO, ZERO];
-    smt.insert(key_b.into(), val_b);
+    let source = "
+    use.std::collections::smt
+    begin
+      exec.smt::set
+    end
+    ";
 
-    // insert a value under key_c which has the same 32-bit prefix as A
-    let raw_c = 0b00000000_00000000_11111111_11111111_00111111_11111111_11111111_11111111_u64;
-    let key_c = RpoDigest::from([ONE, ONE, ONE, Felt::new(raw_c)]);
-    let val_c = [ONE, ONE, ONE, ZERO];
+    let key = LEAVES[0].0;
```

Comment:
Should this be some other key? i.e., the key that currently is associated with an `EMPTY_VALUE`?

Reply:
The tree is empty - we called `Smt::new()` instead of `Smt::with_leaves()`. But I changed it to a different key to avoid the confusion